### PR TITLE
Remove inappropriate ICG test

### PR DESCRIPTION
**Pull request by armenatzoglou**
Commit 9d19664ccbe2b81069249924538f61bba8179b8d ported a test from cdbfast to master, which uses gp_fault_inject mechanism to throw a SIGSEGV.
We remove this test because i) gp_fault_inject works only if USE_TEST_UTILS = 1, while when we test GPDB, we set USE_TEST_UTILS to 0, and ii) throwing segmentation faults in ICG is a wrong approach.

We have created a story to query execution backlog to replace it with unit tests.

### CODE REVIEW

**Comment by armenatzoglou** [comment]
@gcaragea @karthijrk Please take a look if you have time.

**Comment by vraghavan78** [comment]
+1

**Comment by danielgustafsson** [comment]
Agreed, segfaulting on purpose should not be done in ICG. :+1:

File: src/test/regress/expected/qp_query_execution.out [hunk 1]
```diff
@@ -1364,12 +1364,6 @@ select foo_p.b, foo_p.t from foo_p inner join bar on foo_p.a = bar.k  where foo_
 (10 rows)
 
 drop function if exists mytest(integer);
-select gp_fault_inject(12,0);
- gp_fault_inject 
------------------
-               0
-(1 row)
-
```
Comment (gcaragea) [inline src/test/regress/expected/qp_query_execution.out]: I'm pretty sure that the CREATE TABLE below is part of the same test. The JIRA which prompted this test was that there was corrupted strings in the csv log file after a PM reset. 

When we looked at the original test in TINC, the test was doing the following: 
* throwing SEGV using gp_fault_inject to trigger the PM reset
* creating a table
* querrying the log file to check that there was no corrupted log entries

Let's get rid of the CREATE table as well.

**Comment by asimrp** [comment]
:+1:

The function `gp_fault_inject()` is a new way of inject faults for me.  All this while, I was using `gpfaultinjector` Python utility for the same.

File: src/test/regress/expected/qp_query_execution.out (hunk 1 above)
Comment (armenatzoglou) [inline src/test/regress/expected/qp_query_execution.out]: Good point, I removed it. Thanks

**Comment by armenatzoglou** [comment]
Thanks a lot for the review. Pushed to master (23c81a0).

**Pull request closed without merging**